services/manage/app/components/external-links.js

```diff
@@ -0,0 +1,91 @@
+import Component from '@ember/component';
+import { isArray } from '@ember/array';
+import { get, set } from '@ember/object';
+import { debounce } from '@ember/runloop';
+
+export default Component.extend({
+  isModalShowing: false,
+
+  isCreating: false,
+  typeDelay: 750,
+  error: null,
+
+  init() {
+    this._super(...arguments);
+    if (!isArray(this.get('value'))) {
+      this.set('value', []);
+    } else {
+      this.set('value', this.get('value').slice());
+    }
+  },
+
+  /**
+   * Sends the on-change events.
+   * Will only fire when the field is valid.
+   *
+   * @param {string} value
+   * @param {Event} event
+   */
+  async sendOnChange(val) {
+    const value = val.map(link => ({ label: link.label, url: link.url }));
+    const fn = this.get('on-change');
+    if (typeof fn === 'function') {
+      this.set('isChanging', true);
+      try {
+        await fn(value);
+        this.set('changeComplete', true);
+      } catch (e) {
+        this.set('error', e);
+        throw this.get('graphErrors').handle(e);
+      } finally {
+        this.set('isChanging', false);
+      }
+    }
+  },
+
+  /**
+   * Debounces the input onchange event.
+   *
+   * Will immediately fire the change, if dirty, and
+   * will cancel any other pending change events.
+   *
+   * @param {Event} event
+   */
+  debounceChange(event) {
+    this.set('error', null);
+    const value = this.get('value');
+    if (this.get('shouldSave')) {
+      debounce(this, 'sendOnChange', value, event, 1, true);
+    } else {
+      this.sendOnChange(value, event);
+    }
+  },
+
+  actions: {
+    createLink() {
+      this.get('value').pushObject({ label: 'New Link', url: '' });
```

Comment:
The `label` should be an empty string by default... it's more difficult for someone to have to delete `New Link` every time as opposed to just leaving it blank